aiida/orm/nodes/data/structure.py

```diff
@@ -1686,6 +1662,11 @@ def set_pbc(self, value):
             raise ModificationNotAllowed('The StructureData object cannot be modified, it has already been stored')
         the_pbc = get_valid_pbc(value)
 
+        # In order to introduce this consistency check, we would need to change the default value of `pbc`
```

Comment:
I would remove this new commented text here, and instead do this check in the `_validate` method: if the cell is None, then pbc MUST be [False, False, False]